src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs

```diff
@@ -202,6 +202,45 @@ public override float GetDamageMultiplierForBodyPart(BoneBodyPartType bodyPart,
         defenderStunMultiplier = 1f;
     }
 
+    // Consider reworking once https://forums.taleworlds.com/index.php?threads/missioncombatmechanicshelper-getdefendcollisionresults-bypass-strikemagnitudecalculationmodel.459379 is fixed
+    public override bool DecideCrushedThrough(
+        Agent attackerAgent,
+        Agent defenderAgent,
+        float totalAttackEnergy,
+        Agent.UsageDirection attackDirection,
+        StrikeType strikeType,
+        WeaponComponentData defendItem,
+        bool isPassiveUsage)
+    {
+        EquipmentIndex wieldedItemIndex = attackerAgent.GetWieldedItemIndex(Agent.HandIndex.OffHand);
+        if (wieldedItemIndex == EquipmentIndex.None)
+        {
+            wieldedItemIndex = attackerAgent.GetWieldedItemIndex(Agent.HandIndex.MainHand);
+        }
+
+        var weaponComponentData = wieldedItemIndex != EquipmentIndex.None
+            ? attackerAgent.Equipment[wieldedItemIndex].CurrentUsageItem
+            : null;
+        if (weaponComponentData == null
+            || isPassiveUsage
+            || !weaponComponentData.WeaponFlags.HasAnyFlag(WeaponFlags.CanCrushThrough)
+            || strikeType != StrikeType.Swing
+            || attackDirection != Agent.UsageDirection.AttackUp)
+        {
+            return false;
+        }
+
+        float attackerPowerStrikeSkill = 3f * GetSkillValue(attackerAgent.Origin, CrpgSkills.PowerStrike);
```

Comment:
Rename variable to attackerPowerStrikeSkillPower
was a bit confused in Line 241 that we only take the raw attackerPowerStrikeSkill against the defenderDefendPower.